File: task3.pl
#!/usr/bin/perl

use strict;
use warnings;
use LWP::UserAgent;
use JSON;

# Function to fetch variant information from Ensembl API
sub fetch_variant_info {
    my ($variant_id) = @_;

    my $url = "https://rest.ensembl.org/variation/human/$variant_id";
    my $ua = LWP::UserAgent->new;
    my $response = $ua->get($url);

    if ($response->is_success) {
        my $variant_info = decode_json($response->content);
        return $variant_info;
    } else {
        print "HTTP Error: " . $response->status_line . "\n";
        return;
    }
}

# Example variant ID
my $variant_id = "rs56116432";

# Fetch variant information
my $variant_info = fetch_variant_info($variant_id);

if ($variant_info) {
    # Parse the JSON response to extract required information
    my $alleles = $variant_info->{alleles};
    my $location = $variant_info->{mappings}->[0]->{location};
    my $effects = $variant_info->{evidence}->{consequence_types};
    my $genes = $variant_info->{transcript_consequences};

    print "Alleles: $alleles\n";
    print "Location: $location\n";
    print "Effects: @$effects\n";
    print "Genes containing the transcripts:\n";
    foreach my $gene (@$genes) {
        print "- Gene: $gene->{gene_id} Transcript: $gene->{transcript_id}\n";
    }
}

File: task3.py
#!/opt/anaconda3/bin/python3

import requests

# Function to fetch variant information from Ensembl API
def fetch_variant_info(variant_id):
    url = f"https://rest.ensembl.org/variation/human/{variant_id}"
    response = requests.get(url)
    try:
        response.raise_for_status()  # Check for HTTP errors
        variant_info = response.json()  # Parse JSON response
        return variant_info
    except requests.exceptions.HTTPError as err:
        print(f"HTTP Error: {err}")
    except requests.exceptions.RequestException as err:
        print(f"Request Exception: {err}")
    return None

# Example variant ID
variant_id = "rs56116432"

# Fetch variant information
variant_info = fetch_variant_info(variant_id)

if variant_info:
    # Parse the JSON response to extract required information
    alleles = variant_info.get('alleles')
    location = variant_info.get('mappings')[0].get('location')
    effects = variant_info.get('evidence')['consequence_types']
    genes = variant_info.get('transcript_consequences')

    print("Alleles:", alleles)
    print("Location:", location)
    print("Effects:", effects)
    print("Genes containing the transcripts:")
    for gene in genes:
        print("- Gene:", gene.get('gene_id'), "Transcript:", gene.get('transcript_id'))

